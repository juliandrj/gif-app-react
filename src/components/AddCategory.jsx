import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    };
    const onSubmit = (event) => {
        event.preventDefault();
        const cleanInputValue = inputValue?.trim();
        if (cleanInputValue.length <= 2) {
            return;
        }
        onNewCategory(cleanInputValue);
        setInputValue('');
    };
    return (
        <form onSubmit={onSubmit}>
            <input type="text" placeholder="Buscar gifs..." value={inputValue} onChange={onInputChange} />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
