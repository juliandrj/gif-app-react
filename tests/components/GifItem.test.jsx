import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components"

const title = 'TÃ­tulo';
const url = 'http://127.0.0.1/img.jpg';

describe('Tests del componente GifItem', () => {
    test('debe coincidir con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
    test('la imagen debe tener el src y el alt determinado', () => {
        render(<GifItem title={title} url={url} />);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    test('el parrafo debe tener el texto determinado', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});