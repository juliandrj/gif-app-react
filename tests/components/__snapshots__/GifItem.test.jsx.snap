// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests del componente GifItem debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título"
      src="http://127.0.0.1/img.jpg"
    />
    <p>
      Título
    </p>
  </div>
</div>
`;
